Won't start on Android

### Describe the bug

When I run the Avalonia project on an Android device, I get an error

![image](https://github.com/user-attachments/assets/8c727275-360a-46a2-985b-bb28c7d766b1)

App.axaml
![image](https://github.com/user-attachments/assets/d5bc8293-6b06-4980-9ed7-be5a3872eaa7)

App.axaml.cs
![image](https://github.com/user-attachments/assets/66474f1a-2e06-4cd7-933e-3e8974a8a072)

MainActivity
![image](https://github.com/user-attachments/assets/9ab61245-0a49-4dc1-b31b-a2765bea0b14)

But this problem does not occur when I downgrade the Avalonia version from 11.2.0 to 11.1.4

### To Reproduce

Same as above

### Expected behavior

_No response_

### Avalonia version

11.2.0

### OS

Android

### Additional context

_No response_

Same issue here. My observations
* My project and the minimal example above both have the `App.axaml` in a 'shared' project and the andorid part in an own project
* It happens also on Debug builds
* It does not happen after Clean of Solution
* Exception:
```
Avalonia.Markup.Xaml.XamlLoadException: No precompiled XAML found for Xovis.Production.Warehouse.App.Common.App, make sure to specify x:Class and include your XAML file as AvaloniaResource
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Object obj)
   at Xovis.Production.Warehouse.App.Common.App.Initialize() in C:\git\xovis\production-warehouse-app\Xovis.Production.Warehouse.App.Common\App.axaml.cs:line 24
   at Avalonia.AppBuilder.SetupUnsafe()
   at Avalonia.AppBuilder.Setup()
   at Avalonia.AppBuilder.SetupWithLifetime(IApplicationLifetime lifetime)
   at Avalonia.Android.AvaloniaMainActivity.InitializeAvaloniaView(Object initialContent)
   at Avalonia.Android.AvaloniaActivity.OnCreate(Bundle savedInstanceState)
   at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_savedInstanceState) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.App.Activity.cs:line 3082
   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V callback, IntPtr jnienv, IntPtr klazz, IntPtr p0) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 121
```

* When building on CI (first `dotnet build <Solution>`, then `dotnet publish <Android>.csproj --no-restore`, I also get the Exception: 
```
/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018: The "GetAssemblyAttributes" task failed unexpectedly. [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018: System.IO.FileNotFoundException: ${WORKSPACE}/<SharedProject>/obj/Release/net8.0/<SharedProject>.dll [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018:    at System.Diagnostics.FileVersionInfo.GetVersionInfo(String fileName) [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GetAssemblyAttributes.ExecuteCore() [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute() [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]

/usr/share/dotnet/sdk/8.0.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1127,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [${WORKSPACE}/<SharedProject>/<SharedProject>.csproj]
```

--- a/packages/Avalonia/AvaloniaBuildTasks.targets
+++ b/packages/Avalonia/AvaloniaBuildTasks.targets
@@ -133,7 +133,7 @@
     DependsOnTargets="$(CompileAvaloniaXamlDependsOn)"
     Inputs="@(CompileAvaloniaXamlInputs)"
     Outputs="@(CompileAvaloniaXamlOutputs)"
-    Condition="'@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(EnableAvaloniaXamlCompilation) != false AND '$(SkipCompilerExecution)' != 'true'">
+    Condition="'@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(DesignTimeBuild) != true AND $(EnableAvaloniaXamlCompilation) != false">
 
     <!--
       $(IntermediateOutputPath)/Avalonia/references is using from AvaloniaVS for retrieve library references.
@@ -161,7 +161,7 @@
   </Target>
   
   <Target Name="InjectAvaloniaXamlOutput" DependsOnTargets="PrepareToCompileAvaloniaXaml" AfterTargets="CompileAvaloniaXaml" BeforeTargets="CopyFilesToOutputDirectory;BuiltProjectOutputGroup;ComputeResolvedFilesToPublishList"
-          Condition="'@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(EnableAvaloniaXamlCompilation) != false AND '$(SkipCompilerExecution)' != 'true'">
+          Condition="'@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(EnableAvaloniaXamlCompilation) != false">
     <ItemGroup>
       <_AvaloniaXamlCompiledAssembly Include="@(IntermediateAssembly->Metadata('AvaloniaCompileOutput'))"/>
       <IntermediateAssembly Remove="@(IntermediateAssembly)"/>
@@ -192,7 +192,7 @@
           DependsOnTargets="InjectAvaloniaXamlOutput"
           AfterTargets="ComputeIlcCompileInputs"
           BeforeTargets="PrepareForILLink"
-          Condition="$([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '9.0')) AND '@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(EnableAvaloniaXamlCompilation) != false AND '$(SkipCompilerExecution)' != 'true'">
+          Condition="$([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '9.0')) AND '@(AvaloniaResource)@(AvaloniaXaml)' != '' AND $(EnableAvaloniaXamlCompilation) != false">
     <ItemGroup>
       <ManagedBinary Remove="$(IntermediateOutputPath)$(TargetName)$(TargetExt)" />
       <ManagedBinary Include="@(_AvaloniaXamlCompiledAssembly)" />

